test/tests/functional/pbs_modifyresv_hook.py

```diff
@@ -0,0 +1,284 @@
+# option) any later version.
+#
+# OpenPBS is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# PBS Pro is commercially licensed software that shares a common core with
+# the OpenPBS software.  For a copy of the commercial license terms and
+# conditions, go to: (http://www.pbspro.com/agreement.html) or contact the
+# Altair Legal Department.
+#
+# Altair's dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of OpenPBS and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair's trademarks, including but not limited to "PBS™",
+# "OpenPBS®", "PBS Professional®", and "PBS Pro™" and Altair's logos is
+# subject to Altair's trademark licensing policies.
+
+import textwrap
+from tests.functional import *
+
+
+class TestModifyResvHook(TestFunctional):
+    """
+    Tests to verify the reservation begin hook for a confirm standing/advance/
+    degraded reservation once the reservation begins.
+    """
+
+    advance_resv_hook_script = textwrap.dedent("""\
+        import pbs
+        e=pbs.event()
+
+        pbs.logmsg(pbs.LOG_DEBUG, 'Reservation Modify Hook name - %s' %
+                   e.hook_name)
+
+        if e.type == pbs.MODIFYRESV:
+            pbs.logmsg(pbs.LOG_DEBUG, 'Reservation ID - %s' % e.resv.resvid)
+    """)
+
+    def setUp(self):
+        """
+        Create a reservation begin hook and set the server log level.
+        """
+        super(TestModifyResvHook, self).setUp()
+        self.hook_name = 'modifyresv_hook'
+        attrs = {'event': 'modifyresv'}
+        self.server.create_hook(self.hook_name, attrs)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'log_events': 2047})
+
+    @tags('hooks')
+    def test_delete_advance_resv(self):
+        """
+        Testcase to submit and confirm advance reservation, delete the same
+        and verify the modifyresv hook did not run.
+        """
+        self.server.import_hook(self.hook_name,
+                                TestModifyResvHook.advance_resv_hook_script)
+
+        offset = 10
+        duration = 30
+        rid = self.server.submit_resv(offset, duration)
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attrs, id=rid)
+
+        self.server.delete(rid)
+        msg = 'Hook;Server@%s;Reservation ID - %s' % \
+              (self.server.shortname, rid)
+        self.server.log_match(msg, tail=True, interval=1, max_attempts=10,
+                              existence=False)
```

Comment:
The point of the check is that the hook doesn't get triggered on a reservation delete and flagging if that behavior changes.

Reply:
I still don't think this test is necessary.  This hook will only fire when you do a pbs_ralter, not pbs_rdel.  I could see this test for the begin or end hook because it is possible the delete didn't remove the tasks to start/end the reservation (although that would lead to a crash).  I thought I might be too close to the code so I asked someone in our QA organization what she thought, and she agreed with me.  She didn't think this test was necessary.

Reply:
Given the discussion, I'm fine with removing this case.  Will-do.

Reply:
Removed in 00d1d03